src/main/kotlin/com/sourcegraph/cody/autocomplete/CodyAutocompleteManager.kt

```diff
@@ -275,15 +278,43 @@ class CodyAutocompleteManager {
       offset: Int,
       items: List<InlineAutocompleteItem>,
       inlayModel: InlayModel,
-      triggerKind: InlineCompletionTriggerKind
+      triggerKind: InlineCompletionTriggerKind,
+      lookupString: String?,
   ) {
     val defaultItem = items.firstOrNull() ?: return
     val range = getTextRange(editor.document, defaultItem.range)
     val originalText = editor.document.getText(range)
-    val insertTextFirstLine: String = defaultItem.insertText.lines().firstOrNull() ?: ""
+    var insertTextFirstLine: String = defaultItem.insertText.lines().firstOrNull() ?: ""
+    val lineNumber = editor.document.getLineNumber(offset)
+    val caretPositionInLine = offset - editor.document.getLineStartOffset(lineNumber)
+
     val multilineInsertText: String =
         defaultItem.insertText.lines().drop(1).joinToString(separator = "\n")
 
+    if (!lookupString.isNullOrEmpty()) {
+      val lastCommonSuffixCharacterPosition =
+          findLastCommonSuffixElementPosition(
+              originalText.substring(0, caretPositionInLine), lookupString)
+      insertTextFirstLine =
+          insertTextFirstLine.removeRange(lastCommonSuffixCharacterPosition, caretPositionInLine)
+
+      // For each autocomplete item, remove common part with already inserted part of the lookup
+      // element.
+      items.stream().forEach { item ->
+        run {
```

Comment:
Is it necessary to place this code in run? It seems to me that it doesn't make a difference.